grr/server/grr_response_server/gui/ui/store/client_page_facade.ts

```diff
@@ -380,25 +380,34 @@ export class ClientPageStore extends ComponentStore<ClientPageState> {
 
   // An effect to list approvals.
   private readonly listApprovals = this.effect<void>(
-      obs$ => obs$.pipe(
-          switchMapTo(this.selectedClientId$),
-          switchMap(clientId => this.httpApiService.listApprovals(clientId)),
-          map(apiApprovals => apiApprovals.map(translateApproval)),
-          tap(approvals => {
-            this.updateApprovals(approvals);
-          })));
+    obs$ => obs$.pipe(
+      switchMapTo(this.selectedClientId$),
+      switchMap(clientId => this.httpApiService.listApprovals(clientId)),
+      map(apiApprovals => apiApprovals.map(translateApproval)),
+      tap(approvals => {
+        this.updateApprovals(approvals);
+      })));
 
   // An effect to list flows.
   private readonly listFlows = this.effect<void>(
-      obs$ => obs$.pipe(
-          switchMapTo(this.selectedClientId$),
-          exhaustMap(
-              clientId => this.httpApiService.listFlowsForClient(clientId)),
-          map(apiFlows => apiFlows.map(translateFlow)),
-          tap(flows => {
-            this.updateFlows(flows);
-          }),
-          ));
+    obs$ => obs$.pipe(
+      switchMapTo(this.selectedClientId$),
+      exhaustMap(
+        clientId => this.httpApiService.listFlowsForClient(clientId)),
+      map(apiFlows => apiFlows.map(translateFlow)),
+      tap(flows => {
+        this.updateFlows(flows);
+      }),
+    ));
+
+  // An effect to add a label to the selected client
+  readonly addClientLabel = this.effect<string>(
+    obs$ => obs$.pipe(
+      withLatestFrom(this.selectedClientId$),
+      map(([label, clientId]) => {
```

Comment:
In your code, `map` returns an Observable that would send the HTTP POST request, but this inner observable is not subscribed to. Try `switchMap` instead of `map`. Further reading: https://medium.com/@luukgruijs/understanding-rxjs-map-mergemap-switchmap-and-concatmap-833fc1fb09ff

Reply:
Thanks, that was the problem!